src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1200,5 +1386,56 @@ private bool IdentifierIs(string expected, string identifier)
                 identifier,
                 this.configuration.EnableCaseInsensitiveUriFunctionIdentifier ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal);
         }
+
+        /// <summary>
+        /// Validates the existing parsed segments and returns a list of validated segments.
+        /// </summary>
+        /// <returns>List of validated path segments.</returns>
+        private List<ODataPathSegment> CreateValidatedPathSegments()
+        {
+            List<ODataPathSegment> validatedSegments = new List<ODataPathSegment>(this.parsedSegments.Count);
+            for (int index = 0, segmentCount = this.parsedSegments.Count; index < segmentCount; ++index)
+            {
+                CheckDollarEachSegmentRestrictions(index);
+#if DEBUG
+                this.parsedSegments[index].AssertValid();
+#endif
+                validatedSegments.Add(this.parsedSegments[index]);
+            }
+
+            return validatedSegments;
+        }
+
+        /// <summary>
+        /// Per OData 4.01 spec, GET operation and functions may proceed $each, but we are limiting the scope of that spec
+        /// by permitting only ONE action segment to follow $each. As such, at most one $each segment can exist in the URI.
+        /// This function enforces those restrictions and throws if any of them are violated.
+        /// </summary>
+        /// <param name="index">Index of path segment to examine in the list of parsed segments.</param>
+        /// <exception cref="ODataException">Throws if there's a violation of $each restrictions.</exception>
+        /// <remarks>Should the restrictions on the $each be removed, this function can be deleted.</remarks>
+        private void CheckDollarEachSegmentRestrictions(int index)

```

Comment:
@mikepizzo, regarding the single bound action/function following $each, I have this function here that's enforcing that constraint that we decided to impose at the time. It's going to be a little difficult for me to dig deeper into this or make new changes given some time constraints that I have. If there should be further adjustments to this logic, could that be done in a separate PR prior to shipping the next ODL?

Reply:
In line 1434, can you just remove:
       || !(operationSegment.Operations.All(a => a is IEdmAction))
?

---
In reply to: [225763923](https://github.com/OData/odata.net/pull/1110#discussion_r225763923) [](ancestors = 225763923)

Reply:
Just saw your edit from one of the other posts. Yes, removing that condition will allow any single operation to follow $each. Making that change and adjusting tests.